src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -41,14 +53,58 @@ private SymbolNamesOption()
 
             var namesBuilder = PooledDictionary<string, string?>.GetInstance();
             var symbolsBuilder = PooledDictionary<ISymbol, string?>.GetInstance();
+            var wildcardNamesBuilder = PooledDictionary<SymbolKind, PooledDictionary<string, string?>>.GetInstance();
 
             foreach (var symbolName in symbolNames)
             {
                 var parts = getSymbolNamePartsFunc != null
                     ? getSymbolNamePartsFunc(symbolName)
                     : new NameParts(symbolName);
 
-                if (parts.SymbolName.Equals(".ctor", StringComparison.Ordinal) ||
+                var numberOfWildcards = symbolName.Count(c => c == WildcardChar);
+
+                // More than one wildcard, or wildcard char is not last, bail-out.
+                if (numberOfWildcards > 1 ||
+                    (numberOfWildcards == 1 && symbolName[symbolName.Length - 1] != WildcardChar))
+                {
+                    continue;
+                }
+
+                if (numberOfWildcards == 1)
+                {
+                    Debug.Assert(parts.SymbolName[parts.SymbolName.Length - 1] == WildcardChar);
+
+                    if (parts.SymbolName[1] != ':')
```

Comment:
Can this throw an `IndexOutOfRangeException`?

Reply:
I guess it could for some invalid data entry. I'll go defensive.